src/collectors/statsd/statsd.py

```diff
@@ -70,6 +70,7 @@ def get_default_config(self):
     def flush(self):
         for key, value in self.listener_thread.gauges.items():
             self.publish(key, value)
+            del(self.listener_thread.gauges[key])
```

Comment:
Can we make this a call to make on self.listener_thread (`self.listener_thread.clear_gauge(key)`) just so that we arent breaking isolation inside of another class like this.